jupyter_book/commands/__init__.py

```diff
@@ -1,39 +1,53 @@
-"""Defines the commands that the CLI will use."""
+"""Defines the commands that the CLI will use.
+
+IMPORTANT: Top-level imports should be minimised here, to improve CLI responsiveness
```

Comment:
are imports really that expensive? I have always thought that best practice / style was having the imports at the top of a file as a single point of reference.

Reply:
if we do `lazy` imports is there any documentation we should add at the top of the module such as:

```
# {{ function }}:: {{ imports }}
```

Reply:
Normally, thats absolutely the case. But not for CLI, because all these imports have to install everytime you call the CLI and does have a noticeable difference. This is particularly accute, if you think about adding [tab completion](https://github.com/click-contrib/click-completion)

Lazy imports is also what we do in AiiDa: https://github.com/aiidateam/aiida-core/tree/develop/aiida/cmdline/commands

Reply:
> if we do lazy imports is there any documentation 

Personally, I don't think its necessary, and also it would be prone to becoming outdated.
The main thing with the imports, is that we check they all work, which is achieved by the good test coverage

Reply:
Gotcha. I guess in that case I am just wondering how we determine first point of entry for an import? Or do you specify what is needed in each function?

Reply:
Do `imports` respect function scope or are they still globally available?

Reply:
> first point of entry for an import?

not quite sure what you mean by this? If you import the same thing multiple times, that's not a problem since it caches them

you just have to make sure that things are available within the scope you are working, which flake8 will warn about any if something is not available (+ would be picked up in the tests)